src/parser/helper/Language.ts

```diff
@@ -102,6 +109,57 @@ export const fileExtensionToLanguage = new Map([
     ["sh", Language.Bash],
 ]);
 
+/**
+ * Maps supported file extensions to the corresponding programming languages.
+ * For case-sensitive file extensions.
+ */
+const caseSensitiveFileExtensionToLanguage = new Map([
+    ["c", Language.C],
+    ["C", Language.CPlusPlus],
+    ["H", Language.CPlusPlus], // lowercase .h has a special treatment
+]);
+
+/**
+ * Estimates the language of a file based upon the file extension and file path.
+ * @param filePath Path to the file, including the file extension.
+ * @param config Configuration to apply.
+ * @param pathModule ONLY FOR TESTING PURPOSES: overrides the platform-specific path module.
+ */
+export function assumeLanguageFromFilePath(
+    filePath: string,
+    config: Configuration,
+    pathModule = path,
+) {
+    const fileExtension: string = getFileExtension(filePath);
+
+    // Handling of the parse .h as C option:
+    if (fileExtension === "h") {
+        if (config.parseAllAsC) {
+            return Language.C;
+        }
+        if (config.parseSomeAsC.size > 0) {
+            // Use the path relative to the sources path to avoid the unintuitive behaviour
+            // that higher-level folders are evaluated for this:
+            const relativePath = pathModule.relative(config.sourcesPath, filePath);
+            const backwardSlashRelpath = replaceForwardWithBackwardSlashes(relativePath);
+            const relpathSplitted = backwardSlashRelpath.split("\\");
+            for (const pathElement of relpathSplitted) {
+                if (config.parseSomeAsC.has(pathElement)) {
+                    return Language.C;
+                }
+            }
+        }
+        return Language.CPlusPlus;
+    }
+
+    let result = caseSensitiveFileExtensionToLanguage.get(fileExtension);
+    const inLowerCase = fileExtension.toLowerCase();
+    if (result === undefined) {
+        result = fileExtensionToLanguage.get(inLowerCase);
+    }
+    return result;
```

Comment:
I think this lookup logic will most likely be reused in other parts. To extract this into a helper function would make sense.

Reply:
A "look in this map first and if there is no value, look in the second one" helper function, or what do you mean?

Reply:
Well, what is this function actually doing if you would explain it to me? That could be the function name then.

Reply:
The question was more related to the scope of the function to extract here.
I am not sure if a function that first looks in one map and if there is no value checks another map with the key converted to lower case is something that we will ever use outside of this function. Converting to lower case and then looking up a value in one map, perhaps.

Reply:
I have now introduced a helper function `lookupLowerCase`. However, I am not sure if we ever can use that somewhere else and it does only contain two lines of code, so I am unsure if that is a good decision.